tests/modules/test_securitydata.py

```diff
@@ -572,3 +610,311 @@ def test_saved_searches_returns_saved_search_client(
+        file_event_response = response
+        file_event_response.text = FILE_EVENTS_RESPONSE
+        return file_event_response
+
+    @pytest.fixture
+    def file_location(self, mocker):
+        response = mocker.MagicMock(spec=Py42Response)
+        response.status_code = 200
+        response.encoding = None
+        response.__getitem__ = lambda _, key: json.loads(response.text).get(key)
+        file_location_response = response
+        file_location_response.text = FILE_LOCATION_RESPONSE
+        return file_location_response
+
+    @pytest.fixture
+    def find_file_version(self, mocker):
+        response = mocker.MagicMock(spec=Py42Response)
+        response.status_code = 200
+        response.encoding = None
+        response.__getitem__ = lambda _, key: json.loads(response.text).get(key)
+        pds_file_version_response = response
+        pds_file_version_response.text = PDS_FILE_VERSIONS
+        return pds_file_version_response
+
+    @pytest.fixture
+    def file_download(self, mocker):
+        response = mocker.MagicMock(spec=Py42Response)
+        response.status_code = 200
+        response.encoding = None
+        response.__getitem__ = lambda _, key: json.loads(response.text).get(key)
+        download_token_response = response
+        download_token_response.text = "PDSDownloadToken=token"
+        return download_token_response
+
+    def test_stream_file_by_sha256_returns_stream_of_file(
+        self,
+        mocker,
+        security_client,
+        storage_client_factory,
+        microservice_client_factory,
+        file_event_search,
+        file_location,
+        find_file_version,
+        file_download,
+    ):
+
+        security_module = SecurityModule(
+            security_client, storage_client_factory, microservice_client_factory
+        )
+        file_event_client = mocker.MagicMock(spec=FileEventClient)
+        file_event_client.search.return_value = file_event_search
+        file_event_client.get_file_location_detail_by_sha256.return_value = file_location
+        microservice_client_factory.get_file_event_client.return_value = file_event_client
+
+        pds_client = mocker.MagicMock(spec=PreservationDataServiceClient)
+        pds_client.find_file_versions.return_value = find_file_version
+        microservice_client_factory.get_preservation_data_service_client.return_value = pds_client
+
+        storage_node_client = mocker.MagicMock(spec=StoragePreservationDataClient)
+        storage_node_client.get_download_token.return_value = file_download
+        storage_node_client.get_file.return_value = b"stream"
+        microservice_client_factory.create_storage_preservation_client.return_value = (
+            storage_node_client
+        )
+
+        response = security_module.stream_file_by_sha256("shahash")
+        assert response == b"stream"
+
+    def test_stream_file_by_sha256_raises_file_not_found_error(
```

Comment:
It'd be helpful if you include the `condition` as part of the test signature... something like `test_stream_file_by_sha256_when_given_unknown_hash_raises_file_not_found_error()`

Reply:
Changed method name.